### PR TITLE
State Tree Tasks

### PR DESCRIPTION
**Pull request by TheEmidee**
Created many state tree tasks

### CODE REVIEW

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskActivateAbility.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
+﻿#include "StateTree/Tasks/AIExtStateTreeTaskActivateAbility.h"
+
+#include "VisualLogger/VisualLogger.h"
```
Comment (Frickerson) [inline Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskActivateAbility.cpp]: <>

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskApplyGameplayEffect.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        return EStateTreeRunStatus::Failed;
+    }
+
+    if ( instance_data.EffectClass == nullptr )
+    {
+        UE_VLOG( context.GetOwner(), LogStateTree, Error, TEXT( "FAIExtStateTreeTaskApplyGameplayEffect can't apply gameplay effect because no gameplay effect is bound." ) );
+        return EStateTreeRunStatus::Failed;
+    }
+
+    UAbilitySystemComponent * target_asc = instance_data.TargetAbilitySystemComponent;
```
Comment (Frickerson) [inline Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskApplyGameplayEffect.cpp]: auto *

**Review by Fredaly — APPROVED** [review]

File: Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskActivateAbility.h [hunk 3]
```diff
@@ -0,0 +1,71 @@
+﻿#pragma once
+
+#include "GameplayAbilitySpecHandle.h"
```
Comment (Frickerson) [inline Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskActivateAbility.h]: <>

File: Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskMoveTo.h [hunk 4]
```diff
@@ -0,0 +1,101 @@
+﻿#pragma once
+
+#include "AISystem.h"
```
Comment (Frickerson) [inline Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskMoveTo.h]: <>

File: Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskRunEnvironmentQuery.h [hunk 5]
```diff
@@ -0,0 +1,98 @@
+﻿#pragma once
+
+#include "StateTreeEvents.h"
```
Comment (Frickerson) [inline Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskRunEnvironmentQuery.h]: <>

**Review by Frickerson — CHANGES_REQUESTED** [review]

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskActivateAbility.cpp (hunk 1 above)
Comment (TheEmidee) [reply Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskActivateAbility.cpp]: done

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskApplyGameplayEffect.cpp (hunk 2 above)
Comment (TheEmidee) [reply Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskApplyGameplayEffect.cpp]: done

File: Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskActivateAbility.h (hunk 3 above)
Comment (TheEmidee) [reply Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskActivateAbility.h]: done

File: Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskMoveTo.h (hunk 4 above)
Comment (TheEmidee) [reply Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskMoveTo.h]: done

File: Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskRunEnvironmentQuery.h (hunk 5 above)
Comment (TheEmidee) [reply Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskRunEnvironmentQuery.h]: done

**Review by Frickerson — DISMISSED** [review]

**Pull request merged**